File: csharp/Structural/06-Adapter/Dispatcher/JobDispatcher.cs
public class JobDispatcher
{
    private readonly IReadOnlyList<INotifier> _notifiers;

    public JobDispatcher(IEnumerable<INotifier> notifiers)
    {
        _notifiers = notifiers.ToList().AsReadOnly();
    }

    public void DispatchJob(string jobId, string title)
    {
        Console.WriteLine($"\n--- Dispatching Job #{jobId}: {title} ---");

        foreach (var notifier in _notifiers)
            notifier.SendNotification(jobId, $"Your job '{title}' has been assigned.");
    }
}

File: typescript/Structural/06-Adapter/Dispatcher/JobDispatcher.ts
import { INotifier } from "../Notifier/INotifier";

export class JobDispatcher {
  constructor(private readonly notifiers: INotifier[]) {}

  dispatchJob(jobId: string, title: string): void {
    console.log(`\n--- Dispatching Job #${jobId}: ${title} ---`);

    for (const notifier of this.notifiers)
      notifier.sendNotification(
        jobId,
        `Your job '${title}' has been assigned.`,
      );
  }
}
